dev-util/nvidia-cuda-toolkit/nvidia-cuda-toolkit-8.0.44.ebuild

```diff
@@ -0,0 +1,142 @@
+# Copyright 1999-2016 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+inherit check-reqs cuda unpacker versionator
+
+MYD=$(get_version_component_range 1-2)
+
+DESCRIPTION="NVIDIA CUDA Toolkit (compiler and friends)"
+HOMEPAGE="https://developer.nvidia.com/cuda-zone"
+SRC_URI="https://developer.nvidia.com/compute/cuda/${MYD}/prod/local_installers/cuda_${PV}_linux-run -> cuda_${PV}_linux.run"
+SLOT="0/${PV}"
+LICENSE="NVIDIA-CUDA"
+KEYWORDS="-* ~amd64 ~amd64-linux"
+IUSE="debugger doc eclipse profiler"
+
+DEPEND=""
+RDEPEND="${DEPEND}
+	>=sys-devel/gcc-4.7[cxx]
+	>=x11-drivers/nvidia-drivers-367.44[uvm]
+	debugger? (
+		sys-libs/libtermcap-compat
+		sys-libs/ncurses[tinfo]
+		)
+	eclipse? ( >=virtual/jre-1.6 )
+	profiler? ( >=virtual/jre-1.6 )"
+
+S="${WORKDIR}"
+
+QA_PREBUILT="opt/cuda/*"
+
+CHECKREQS_DISK_BUILD="3500M"
+
+pkg_setup() {
+	# We don't like to run cuda_pkg_setup as it depends on us
+	check-reqs_pkg_setup
+}
+
+src_unpack() {
+	unpacker
+	unpacker run_files/cuda-linux*.run
+}
+
+src_prepare() {
+	local cuda_supported_gcc
+
+	cuda_supported_gcc="4.7 4.8 4.9 5.3"
+
+	sed \
+		-e "s:CUDA_SUPPORTED_GCC:${cuda_supported_gcc}:g" \
+		"${FILESDIR}"/cuda-config.in > "${T}"/cuda-config || die
+
+	default
+}
+
+src_install() {
+	local i j
+	local remove="doc jre run_files install-linux.pl "
+	local cudadir=/opt/cuda
+	local ecudadir="${EPREFIX}"${cudadir}
+
+	# dodoc doc/*txt
+	if use doc; then
+		dodoc doc/pdf/*
+		dohtml -r doc/html/*
```

Comment:
'messes up file structure' -- sounds like you were doing something wrong. `HTML_DOCS` doesn't do anything except for calling `dodoc -r`…